lessons/durablefunctions/chaining-ts.md

```diff
@@ -0,0 +1,631 @@
+# Azure Durable Functions - Introduction & Chaining
+
+## Goal 🎯
+
+The goal of this lesson is to give you an introduction into Azure Durable Functions including a first durable function that chains two functions calls. In addition we will take a look into some features of Durable Functions that help you when working with durable functions.
+
+This lessons consists of the following exercises:
+
+|Nr|Exercise
+|-|-
+|0| [Introduction to Azure Durable Functions](#0-introduction-to-azure-durable-functions)
+|0.1| [Functions and Chaining](#01-functions-and-chaining)
+|0.2| [Solution via Durable Functions](#02-solution-via-durable-functions)
+|0.3| [Mechanics of Durable Functions](#03-mechanics-of-durable-functions)
+|0.4| [Prerequisites for local development](#04-prerequisites-for-local-development)
+|1| [Creating a Function App project for a Durable Function](#1-creating-a-function-app-project-for-a-durable-function)
+|1.1| [The Client Function](#11-the-client-function)
+|1.2| [The Orchestrator Function](#12-the-orchestrator-function)
+|1.3| [The Activity Function](#13-the-activity-function)
+|1.4| [The First Execution](#14-the-first-execution)
+|2| [Implementing a "Real-World" Scenario](#2-implementing-a-"real-world"-scenario)
+|2.1| [Basic Setup of the Scenario](#21-basic-setup-of-the-scenario)
+|2.2| [Implementation of the Activity Function `GetPoDetails`](#22-implementation-of-the-activity-function-`GetPoDetails`)
+|2.3| [Implementation of the Activity Function `GetPgText`](#23-implementation-of-the-activity-function-`getpgtext`)
+|2.4| [Implementation of the Orchestrator Function](#24-implementation-of-the-orchestrator-function)
+|2.5| [Test the Implementation](#25-test-the-implementation)
+|3| [Retries - Dealing with Temporal Errors](#3-retries---dealing-with-temporal-errors)
+|4| [Racing Conditions - Dealing with Timeouts](#4-racing-conditions---dealing-with-timeouts)

```

Comment:
4 becomes 5. Good idea to introduce the durable timer! I would rename this to Dealing with timeouts with a circuit breaker. Because racing conditions has a different meaning (more related to threading etc). Perhaps we can even link to this pattern: https://docs.microsoft.com/en-us/azure/architecture/patterns/circuit-breaker.

Reply:
Done